src/views/Housing/components/RDView/components/RoomRanges/index.jsx

```diff
@@ -149,132 +137,192 @@ const RoomRanges = () => {
+            >
+              <MenuItem value="BRO">Bromley</MenuItem>
+              <MenuItem value="FER">Ferrin</MenuItem>
+              <MenuItem value="EVN">Evans</MenuItem>
+              <MenuItem value="WIL">Wilson</MenuItem>
+              <MenuItem value="CHA">Chase</MenuItem>
+              <MenuItem value="TAV">Tavilla</MenuItem>
+              <MenuItem value="FUL">Fulton</MenuItem>
+              <MenuItem value="NYL">Nyland</MenuItem>
+              <MenuItem value="GRA">Grace</MenuItem>
+              <MenuItem value="MCI">MacInnis</MenuItem>
+              <MenuItem value="CON">Conrad</MenuItem>
+              <MenuItem value="RID">Rider</MenuItem>
+            </Select>
+          </FormControl>
+          <TextField
+            label="Room Start"
+            value={roomStart}
+            onChange={(e) => setRoomStart(e.target.value)}
+            fullWidth
+            margin="normal"
+          />
+          <TextField
+            label="Room End"
+            value={roomEnd}
+            onChange={(e) => setRoomEnd(e.target.value)}
+            fullWidth
+            margin="normal"
+          />
+        </CardContent>
+        <CardActions>
+          <Button variant="contained" onClick={addRoomRange}>
+            Save Range
+          </Button>
+        </CardActions>
+      </Card>
 
-      <Button variant="contained" onClick={addRoomRange} sx={{ mt: 2 }}>
-        Save Range
-      </Button>
+      <Card variant="outlined" sx={{ mb: 3 }}>
+        <CardContent>
+          <Typography variant="h6">Room Ranges</Typography>
+          <Typography variant="body1" gutterBottom>
+            The list below shows the current room ranges. Click on a range to select it, then use
+            "Remove" to delete it.
+          </Typography>
+          <List>
+            {showList ? (
+              roomRanges.length > 0 ? (
+                roomRanges.map((range, index) => (
+                  <ListItem
+                    key={index}
+                    onClick={() => setSelectedRoomRange(index)}
+                    sx={{
+                      display: 'flex',
+                      alignItems: 'center',
+                      gap: 1,
+                      cursor: 'pointer',
+                      backgroundColor:
+                        selectedRoomRange === index ? 'rgba(0, 0, 0, 0.1)' : 'transparent',
+                      '&:hover .remove-btn': {
+                        borderColor: 'black',
+                        color: 'black',
+                      },
+                    }}
+                  >
+                    <Box>
+                      {range.Hall_ID}: {range.Room_Start} - {range.Room_End}
+                    </Box>
+                    <Button
+                      variant="outlined"
+                      color="secondary"
+                      size="small"
+                      onClick={(e) => {
+                        e.stopPropagation(); // Prevent triggering list item click
```

Comment:
What does line 234 do when you use `e.stopPropagation` ? I am a little confused on what it is preventing.

Reply:
helps to make sure the event does not trigger anything inadvertently